src/library/scala/runtime/ClassValueCompat.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.runtime;
+
+public abstract class ClassValueCompat<T> {
+
+    private static final boolean classValueAvailable = checkClassValueAvailability();
+
+    private static boolean checkClassValueAvailability() {
+        try {
+            Class.forName("java.lang.ClassValue", false, ClassValueCompat.class.getClassLoader());
```

Comment:
maybe use Object's class loader, as the two should be defined in the same library?

```suggestion
            Class.forName("java.lang.ClassValue", false, Object.class.getClassLoader());
```